File: lib/src/models/trending_category.dart
import 'package:json_annotation/json_annotation.dart';

import 'search_result_of_trending_entry.dart';

part 'trending_category.g.dart';

@JsonSerializable()
class TrendingCategory{
	
	TrendingCategory();

	factory TrendingCategory.fromJson(Map<String, dynamic> json) => _$TrendingCategoryFromJson(json);

	@JsonKey(name:'categoryName')
	String categoryName;
	@JsonKey(name:'entries')
	SearchResultOfTrendingEntry entries;
	@JsonKey(name:'categoryId')
	String categoryId;

	
	
	Map<String, dynamic> toJson() => _$TrendingCategoryToJson(this);
}

File: lib/src/models/TrendingCategory.cs
using System.Runtime.Serialization;

namespace BungieNetApi.Model {
	[DataContract]
	public class TrendingCategory{

		[DataMember(Name="categoryName", EmitDefaultValue=false)]
		public string CategoryName { get; set; }

		[DataMember(Name="entries", EmitDefaultValue=false)]
		public SearchResultOfTrendingEntry Entries { get; set; }

		[DataMember(Name="categoryId", EmitDefaultValue=false)]
		public string CategoryId { get; set; }


		public override bool Equals(object input)
        {
            return this.Equals(input as TrendingCategory);
        }

		public bool Equals(TrendingCategory input)
		{
			if (input == null) return false;

			return
				(
                    CategoryName == input.CategoryName ||
                    (CategoryName != null && CategoryName.Equals(input.CategoryName))
                ) &&
				(
                    Entries == input.Entries ||
                    (Entries != null && Entries.Equals(input.Entries))
                ) &&
				(
                    CategoryId == input.CategoryId ||
                    (CategoryId != null && CategoryId.Equals(input.CategoryId))
                ) ;
		}

		/* 
		public override int GetHashCode()
		{
			unchecked // Overflow is fine, just wrap
			{
				int hashCode = 41;
				hashCode = hashCode * 59 + this.CategoryName.GetHashCode();
				hashCode = hashCode * 59 + this.Entries.GetHashCode();
				hashCode = hashCode * 59 + this.CategoryId.GetHashCode();
				return hashCode;
			}
		}*/
	}
}

